### PR TITLE
Device fields #25

### PR DESCRIPTION
**Pull request by CityVibes**
Adding missing fields that now are visible in getDevice response.
Also updated example app to include separate activity which will display device information.

@jensck SimpleDevice and CompleteDevice now has even more parameters in constructor, any suggestions? Perhaps build pattern would make some sense?

### CODE REVIEW

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java [hunk 1]
```diff
@@ -119,6 +135,9 @@ public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(deviceType.name());
         dest.writeInt(platformId);
         dest.writeInt(productId);
+        dest.writeString(ipAddress);
```
Comment (jensck) [inline cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java]: Are these nullable fields?  If so, we should be using `writeValue()` and `readValue()`.

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java [hunk 2]
```diff
@@ -135,6 +156,10 @@ public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(deviceType.name());
         dest.writeInt(platformId);
         dest.writeInt(productId);
+        Parcelables.writeBoolean(dest, cellular);
+        dest.writeString(imei);
```
Comment (jensck) [inline cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java]: Same thing here as elsewhere: if these are nullable, we should be using `read/writeValue()`

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java [hunk 3]
```diff
@@ -33,6 +33,10 @@
     final String status;
     final String name;
     final boolean isConnected;
+    final boolean cellular;
```
Comment (jensck) [inline cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java]: if these are nullable (and I bet they are, e.g.: IMEI on a Photon), the fields should be marked as such

File: cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java (hunk 2 above)
Comment (jensck) [reply cloudsdk/src/main/java/io/particle/android/sdk/cloud/DeviceState.java]: Looks like I did that wrong on the `version` field, too.  Might be worth looking for all the nullable fields on these parcelables and applying `read/writeValue()` as needed.

**Review by jensck — CHANGES_REQUESTED** [review]
Need to accommodate the `Parcelable` stuff (if you can't see them: github is calling the comments "outdated" because I put the comments on the commit level rather than that final files)

Also yes, I agree: a builder would be a really good idea, particularly now that we have multiple fields of the same type, one right after the other.

**Comment by CityVibes** [comment]
@jensck Not sure which fields are nullable though, pretty much every field could be depending on Api, right?

**Comment by jensck** [comment]
@CityVibes technically, yes...  ugh.  Yeah, it's probably worth doing that for every object field. :(

It's not an issue ATM, but it should be fixed soon.  If you want, you can just file an issue on it and fix it on another PR.

**Review by jensck — APPROVED** [review]

**Comment by jensck** [comment]
@CityVibes If you want to do the null handling or adding a builder here, go for it; if you want to do them on other PRs and merge what's here now, I'm fine with that, too.

**Pull request merged**